arch/x86/core/x86_mmu.c

```diff
@@ -32,9 +32,18 @@ BUILD_ASSERT(PHYS_RAM_ADDR + PHYS_RAM_SIZE - 1ULL <=
 
 /* Mark text and rodata as read-only.
  * Userspace may read all text and rodata.
+ *
+ * If building with gdb support text will be read/write
  */
+
+#ifdef CONFIG_GDB_SUPPORT
+#define MMU_TEXT_FLAGS (Z_X86_MMU_US | Z_X86_MMU_RW)
+#else
+#define MMU_TEXT_FLAGS Z_X86_MMU_US
+#endif
+
 MMU_BOOT_REGION(&_image_text_start, &_image_text_size,
-		Z_X86_MMU_US);
+		MMU_TEXT_FLAGS);
```

Comment:
why is text read/write? is it because we need to inject trap instructions?

Reply:
yep !